nemo_skills/code_execution/sandbox.py

```diff
@@ -315,13 +339,18 @@ def update_fn(map_to_future, line_dict):
                     line_dict = json.loads(file_line)
                     if not line_dict:  # can be empty for incomplete generations
                         continue
-                    gt_answer = line_dict["expected_answer"]
+                    gt_answer = line_dict.get("expected_answer", "")
```

Comment:
Let's move this inside "if" and still use ["expected_answer"] for the python part to avoid accidentally setting it to the default value, when it's required. And for lean part you can just add it to line_dict with an empty string to be able to reuse map_to_future functionality. Probably will redesign to make it cleaner in the future